docs/rest.md

```diff
@@ -0,0 +1,623 @@
+
+For a standard synchronous operation, the following JSON object is
+returned:
+
+```javascript
+{
+ "result": {},               // Extra resource/action specific data
+ "status": "OK",
+ "status_code": 200,
+ "type": "sync"
+}
+```
+
+The HTTP code must be 200.
+
+### Background operation
+
+When a request results in a background operation, the HTTP code is set
+to 202 (Accepted) and the Location HTTP header is set to the operation
+URL.
+
+The body is a json object with the following structure:
+
+```javascript
+{
+ "result": {
+   "resource": "/1.0/operations/[uuid]",     // see below
+   "status": "running",
+   "created_at": "..."                       // and other operation fields
+ },
+ "status": "Accepted",
+ "status_code": 202,
+ "type": "async"
+}
+```
+
+The response body is mostly provided as a user friendly way of seeing
+what's going on without having to pull the target operation; all
+information in the body can also be retrieved from the background
+operation URL.
+
+### Error
+
+There are various situations in which something may immediately go
+wrong, in those cases, the following return value is used:
+
+```javascript
+{
+ "result": {},            // may contain more details, for debugging
+ "status": "Bad Request", // text description of status_code
+ "status_code": 400,      // or 401, etc. (same as HTTP code)
+ "type": "error"
+}
+```
+
+HTTP code must be one of of 400, 401, 403, 404, 409, 412 or 500.
+
+### Timestamps
+
+Timestamps are presented in µs since the epoch UTC, formatted as a decimal
+string. For example, `"1234567891234567"` represents
+`2009-02-13T23:31:31.234567`.
+
+## /
+### GET
+
+* Description: List of supported APIs
+* Authorization: guest
+* Operation: sync
+* Return: list of supported API endpoint URLs (by default `["/1.0"]`)
+
+
+## /1.0
+### GET
+
+* Description: Server configuration and environment information
+* Authorization: guest
+* Operation: sync
+* Return: Dict with the operating system's key values.
+
+#### Sample result:
+
+```javascript
+{
+ "default_channel": "edge",
+ "flavor": "core",
+ "api_compat": "0",           // increased on minor API changes
+ "release": "15.04",
+ "store": "store-id"          // only if not default
+}
+```
+
+## /1.0/packages
+### GET
+
+* Description: List of packages
+* Authorization: trusted
+* Operation: sync
+* Return: list of URLs for packages this Ubuntu Core system can handle.
+
+The result is a JSON object with a packages key; its value is itself a
+JSON object whose keys are qualified package names (e.g.,
+hello-world.canonical), and whose values describe that package.
+
+Sample result:
+
+```javascript
+{
+ "packages": {
+   "dd.canonical": {
+     "description": "A description",
+     "download_size": "23456",
+     "icon": "/icons/dd.png",
+     "installed_size": "-1",          // always -1 if not installed
+     "name": "dd",
+     "origin": "canonical",
+     "resource": "/1.0/packages/dd.canonical",
+     "status": "not installed",
```

Comment:
In our call we agreed to rename this to "available", but we need to confirm with people in the team whether this is indeed the intended meaning of that state.